src/F2_autoreport.py

```diff
@@ -645,6 +645,35 @@ def create_app(results_json):
     # Rename the timestamp column from 'index' to 'timestamp'
     df_all_demands = df_all_demands.rename(columns={"index": "timestamp"})
 
+    # Collect the keys of various resources (PV, Wind, etc.)
+    resources = results_json[ENERGY_PRODUCTION]
+
+    # List of resources (includes DSOs, which need to be removed)
+    res_keys = list(resources.keys())
+    for res in res_keys:
+        if 'DSO_' in res:
```

Comment:
is it a general treatment or it only applies to the pilot case?

Reply:
If the user follows the naming convention where they add the grid related resources with 'DSO_' in the label, then it would work. Otherwise, I'm not sure how we identify and remove the grid related resources. I asked Martha about it in one of above comments.